Automattic/jetpack
# Custom CSS: update Settings UI to recommend Global Styles over Custom CSS for Block-Based Themes

Fixes https://github.com/Automattic/wp-calypso/issues/71130

## Proposed changes:

This PR introduces a new feature to the Jetpack plugin to detect if the WordPress site has the core site editor feature enabled. This information is stored in the global state tree and can be accessed via the getIsCoreSiteEditorEnabled function. The detection mechanism is used to adjust the Jetpack settings, particularly under Writing settings.

Moreover, the PR makes a contextual adjustment to the user interface. The Jetpack > Settings > Writing > Custom CSS toggle is now conditionally rendered based on whether the site uses a block-based theme or not. For sites using a block-based theme, this toggle is removed, aligning with the concept that such themes inherently offer greater flexibility for customization and thus the additional CSS settings might not be necessary.

This PR involves changes in multiple files, including state reducers, the writing settings component, and theme enhancements, to ensure the changes are effectively integrated.

This is classified as a minor enhancement due to its impact on improving the user interface and experience when interacting with block-based themes.

The change should be transparent to the end-user and help streamline the interface by removing potentially redundant controls for sites using block-based themes.

See some screenshots in #31601. 

### Other information:


## Jetpack product discussion

See discussion in the PR below, and in #31601.

## Does this pull request change what data or activity we track or use?

* Yes, it adds a Tracks event when you click on the link to access Global Styles, matching the existing link to the customizer.

## Testing instructions:

1. Start with a site connected to WordPress.com, either self-hosted on WoA, with a classic theme like Twenty Twenty One.
2. Go to Jetpack > Settings > Writing
    * You should be able to turn the Custom CSS feature on and off.
    * Leave the feature on.
3. Now go to Appearance > Themes and switch to the Twenty Twenty Three theme.
4. Go back to Jetpack > Settings > Writing.
    * You should still have the option to turn the feature off, but a notice now recommends Global styles instead of the Customizer.
5. Turn the feature off
    * The toggle should disappear. A notice should remain, explaining that Global Styles are the way to go now.
